spacy/ml/tb_framework.pyx

```diff
@@ -249,9 +249,11 @@ cdef list _parse_batch(CBlas cblas, TransitionSystem moves, StateC** states,
     cdef np.ndarray step_actions
 
     scores = []
-    while sizes.states >= 1:
+    while sizes.states >= 1 and (actions is None or actions):
```

Comment:
I'm finding the combined condition hard to read and somewhat implicit. Perhaps 
```
actions is None or len(actions)>0
``` 
to make it more explicit that we're guarding against an empty list?

Reply:
Fixed.